amdsharktuner/tests/constraint_generator_test.py

```diff
@@ -438,8 +440,104 @@ def test_adjust_problem_size_for_pipeline(
         matmul_size=conv_size,
         dispatch_kind=common.DispatchKind.conv,
         pipeline_options_search_space=pipeline_options_space,
-        codegen_pipeline=taf_pipeline,
+        codegen_pipeline=tile_and_fuse_pipeline,
     )
     assert pipeline_options_space.use_igemm_convolution == [True]
     assert conv_size.K == [4608]
     assert conv_dims.k == [4]
+
+
+def test_adjust_problem_size_for_pipeline_with_igemm_details(
+    tuner_ctx: common.TunerContext,
+) -> None:
+    """Test adjust_problem_size_for_pipeline with IGEMM details from the binding."""
+    context = tuner_ctx.mlir_ctx
+    f16 = tuner_ctx.type.f16
+    f32 = tuner_ctx.type.f32
+
+    input_shape = (2, 32, 32, 128)
+    kernel_shape = (3, 3, 128, 256)
+    output_shape = (2, 30, 30, 256)
+
+    with ir.Location.unknown(context):
+        module = ir.Module.create()
+        build_func_with_conv2d_nhwc_hwcf(
+            module=module,
+            input_shape=input_shape,
+            kernel_shape=kernel_shape,
+            output_shape=output_shape,
+            input_type=f16,
+            kernel_type=f16,
+            output_type=f32,
+        )
+
+        root_op_list = iree_codegen.get_tuner_root_ops(module)
+        assert len(root_op_list) == 1
+        root_op = root_op_list[0]
+
+        parser = dispatch_parser.ConvolutionOpInterfaceParser(root_op, tuner_ctx)
+        conv_op_info = parser.get_op_info()
+        assert isinstance(conv_op_info, dispatch_parser.ConvolutionOpInfo)
+        assert (
+            conv_op_info.igemm_details is not None
+        ), "IGEMM details should be available for NHWC conv"
+
+        conv_dims = common.ContractionDimensions(
+            m=list(conv_op_info.dims.m),
+            n=list(conv_op_info.dims.n),
+            k=list(conv_op_info.dims.k),
+            batch=list(conv_op_info.dims.batch),
+        )
+        conv_size = common.ContractionSizes(
+            M=list(conv_op_info.matmul_size.M),
+            N=list(conv_op_info.matmul_size.N),
+            K=list(conv_op_info.matmul_size.K),
+            B=list(conv_op_info.matmul_size.B),
```

Comment:
Would it help to assert that what the returned dimensions and sizes are? I think it won't ever change, so we don't have to worry about these getting out of sync with the compiler.